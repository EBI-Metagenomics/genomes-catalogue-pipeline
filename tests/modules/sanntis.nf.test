nextflow_process {

    name "Test Process SANNTIS"
    script "modules/sanntis.nf"
    process "SANNTIS"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
                catalogue_name = "catdog"
                catalogue_version = "1.3"
            }
            process {
                """
                input[0] = channel.from([
                    tuple(
                        "MGYG000000001",
                        file("$baseDir/tests/modules/fixtures/sanntis/MGYG000000001_InterProScan.tsv"),
                        file("$baseDir/tests/modules/fixtures/sanntis/MGYG000000001.gbk")
                    )
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
