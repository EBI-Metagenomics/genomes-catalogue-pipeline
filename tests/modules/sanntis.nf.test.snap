{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "MGYG000000001",
                        "MGYG000000001_sanntis.gff:md5,90cc3823fd29a2a6d574c1be984ba11d"
                    ]
                ],
                "sanntis_gff": [
                    [
                        "MGYG000000001",
                        "MGYG000000001_sanntis.gff:md5,90cc3823fd29a2a6d574c1be984ba11d"
                    ]
                ]
            }
        ],
        "timestamp": "2024-01-11T22:25:32.012275"
    }
}
